Support serializing single fields without outer Avro Record or Json object (#2800)
confluentinc/ksql: docs/installation/server-config/config-reference.rst

`KSQL Server image <https://hub.docker.com/r/confluentinc/cp-ksql-server/>`__ is
``KSQL_KSQL_FUNCTIONS_SUBSTRING_LEGACY_ARGS``.

.. _ksql_persistence_ensure_value_is_struct:

---------------------------------------

> ksql.persistence.ensure.value.is.struct

Reviewer: perhaps `ksql.serialization.value.ensure.struct` - that way we can group under `serialization` and `serialization.value`

Author: Renamed in recent push...

`KSQL Server image <https://hub.docker.com/r/confluentinc/cp-ksql-server/>`__ is
``KSQL_KSQL_FUNCTIONS_SUBSTRING_LEGACY_ARGS``.

.. _ksql_persistence_ensure_value_is_struct:

---------------------------------------
ksql.persistence.ensure.value.is.struct
---------------------------------------

> Controls how KSQL will serialize a value whose schema contains only a single column.

Reviewer 2: ```suggestion
Controls how KSQL serializes a value whose schema contains only a single column.
```

> When set to the default ``true`` KSQL will serialize the column value nested with a JSON object or

Reviewer 2: ```suggestion
When set to the default value, ``true``, KSQL serializes the column value nested with a JSON object or
```

> Avro record, depending on the format in use. When set to ``false`` KSQL will persist the column

Reviewer 2: ```suggestion
Avro record, depending on the format in use. When set to ``false``, KSQL persists the column
```

value without any nesting.

For example, consider the statement:

.. code:: sql

    CREATE STREAM y AS SELECT f0 FROM x;

> The statement selects a single field as the value of stream ``y``. Let us assume that ``f0`` has the

Reviewer 2: ```suggestion
The statement selects a single field as the value of stream ``y``. If ``f0`` has the
```

> integer value ``10``.

Reviewer 2: ```suggestion
integer value ``10``,
```

> 

Reviewer 2: ```suggestion
```

> With ``ksql.persistence.ensure.value.is.struct`` set to ``true`` the JSON format would persist the

Reviewer 2: ```suggestion
with ``ksql.persistence.ensure.value.is.struct`` set to ``true``, the JSON format persists the
```

> value within a JSON object as it would if the value was to have more fields:

Reviewer 2: ```suggestion
value within a JSON object, as it would if the value had more fields:
```

.. code:: json

    {
       "F0": 10
    }

> With ``ksql.persistence.ensure.value.is.struct`` set to ``false`` the JSON format would persist the

Reviewer 2: ```suggestion
With ``ksql.persistence.ensure.value.is.struct`` set to ``false``, the JSON format persists the
```

single field's value as a JSON number: ``10``.

> The ``AVRO`` format can also be controlled by setting ``ksql.persistence.ensure.value.is.struct``.

Reviewer 2: ```suggestion
The ``AVRO`` format can also be controlled by setting ``ksql.persistence.ensure.value.is.struct``,
```

> In this case the setting controls whether the fields value is written nested within an Avro record,

Reviewer 2: ```suggestion
which controls whether or not the field's value is written as nested within an Avro record.
```

> or not.

Reviewer 2: ```suggestion

```

> .. note:: The ``DELIMITED`` format is unaffected by the setting as it has no concept of an outer

Reviewer 2: ```suggestion
.. note:: The ``DELIMITED`` format is  not affected by the `ksql.persistence.ensure.value.is.struct`` setting,
```

>           record or structure.

Reviewer 2: ```suggestion
          because it has no concept of an outer record or structure.
```
